sdk/identity/identity/src/credentials/azurePipelinesServiceConnectionCredential.ts

```diff
@@ -0,0 +1,165 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import { AccessToken, GetTokenOptions, TokenCredential } from "@azure/core-auth";
+import { ClientAssertionCredential } from "./clientAssertionCredential";
+import { CredentialUnavailableError } from "../errors";
+import { credentialLogger } from "../util/logging";
+import { checkTenantId } from "../util/tenantIdUtils";
+import {
+  createDefaultHttpClient,
+  createHttpHeaders,
+  createPipelineRequest,
+} from "@azure/core-rest-pipeline";
+import { AzurePipelinesServiceConnectionCredentialOptions } from "./azurePipelinesServiceConnectionCredentialOptions";
+
+const credentialName = "AzurePipelinesServiceConnectionCredential";
+
+const logger = credentialLogger(credentialName);
+
+export class AzurePipelinesServiceConnectionCredential implements TokenCredential {
+  private client: ClientAssertionCredential | undefined;
+  private serviceConnectionId: string | undefined;
+
+  /**
+   * AzurePipelinesServiceConnectionCredential supports Microsoft Entra Workload ID on Kubernetes.
+   *
+   * @param options - The identity client options to use for authentication.
+   */
+  constructor(
+    clientId: string,
+    tenantId: string,
+    serviceConnectionId: string,
+    options?: AzurePipelinesServiceConnectionCredentialOptions
+  ) {
+    checkTenantId(logger, tenantId);
+    logger.info(
+      `Invoking AzurePipelinesServiceConnectionCredential with tenant ID: ${tenantId}, clientId: ${clientId} and service connection id: ${serviceConnectionId}`
```

Comment:
Is this all public information that can be logged? Just checking 👍

Reply:
yes, checked that with service team as well. It's safe to log these